discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -757,6 +756,73 @@ impl FeedbackScope for SqliteStorage {
         tx.commit().await?;
         Ok(document)
     }
+
+    async fn update_time_spent(
+        &self,
+        document: document::Id,
+        view_mode: ViewMode,
+        duration: Duration,
+    ) -> Result<TimeSpentDocumentView, Error> {
+        let view_time_ms = u32::try_from(duration.as_millis()).ok().unwrap_or(u32::MAX);
+
+        let mut tx = self.pool.begin().await?;
+
+        sqlx::query(
+            "INSERT INTO ViewTimes(documentId, viewMode, viewTimeMs) VALUES (?, ?, ?)
+                ON CONFLICT DO UPDATE SET viewTimeMs = viewTimeMs + excluded.viewTimeMs;",
```

Comment:
❓ if `viewTimeMs` already is (close to) an `u32::MAX`, doesn't this overflow/error when adding the `excluded.viewTimeMs`? it seems that we need a minimum here as well
```suggestion
                ON CONFLICT DO UPDATE SET viewTimeMs = min(viewTimeMs + excluded.viewTimeMs, 4294967295);",
```

Reply:
as the internal representation goes up to `i64` this doesn't really, but adding it doesn't hurt either I guess